### ISSUE TITLE
Add support for project dependencies

### ISSUE THREAD

**Issue by ewilderj** [ewilderj/doap#1]
A need for expression of dependency information in DOAP has been expressed a couple of times on the mailing list [eg. 1,2]. As a start, the attached patch adds dependency support to the DOAP ontology definition. A property 'dependency' is added in the way that's (AFAICS) most advocated on-list. This means that a doap:Project can have a doap:dependency on another doap:Project.

Ported from Trac issues

**Comment by exhuma** [comment]
How about dependencies on other resources? Like databases, or external/remote services?

The diff states that the range is another `doap:Project` instance. Technically, one could create another `doap:Project` entry for the database (for example), and then link them like this. Would this be a recommended way to do this? Personally, I don't necessarily see a database as a "Project", but   I could live with it.

Also, when creating a doap instance for a database or remote service it would be interesting to flag these. I found the "category" property, but the schema does not define a range for that. Is that a "trove" classifier?

**Comment by kjetilk** [comment]
BTW, Toby Inkster has made a separate vocabulary for this: http://ontologi.es/doap-deps#
We use it operationally when building CPAN packages. I think it does the job well.

**Comment by ewilderj** [comment]
@kjetilk thanks for that notification. I wonder if it would make sense to incorporate this vocabulary into DOAP? If Toby or you have interest, I would welcome there being another person or two keeping DOAP healthy. My current day job and spare time don't allow me much focus on it.

**Comment by kjetilk** [comment]
Toby isn't much involved anymore either, but it seems I am in it for the long haul, but you can never know :-) So, I would happily accept co-maintainance of DOAP! 

I am divided as to whether it makes sense to incorporate doap-deps into DOAP proper, as already deployed code depends on it, and I tend to prefer small vocabularies that do one job and does it well, and we have that now, I think. 

But OTOH I realize that the world seems to agree that stuffing everything in one namespace is better (schema.org :-) ).

**Comment by ewilderj** [comment]
@kjetilk just to tidy this thread up, I added you as a Collaborator in GitHub. We should probably figure out how to have a lightweight collaboration that works for us both. I know I have several outstanding tasks I need to do.

**Comment by kjetilk** [comment]
Yeah, I have noticed, @ewilderj ! Thanks a lot! The main issue I have now is that I don't know how to initiate rebuilding the namespace document, which is needed for changes to have impact, I think.

In other good news, @tobyink have resumed some activity.

**Comment by ewilderj** [comment]
Yep, that's the thing I need to do. Basically, it's on one of my servers and I need to run a Makefile. In the near term I can do that, but medium term we might need to find a way of hosting it that doesn't require it, or share your SSH key to me so you can do it too? Should probably take this conversation out of the issue…

**Comment by kjetilk** [comment]
On Wednesday 22. February 2017 15.24.13 Edd Wilder-James wrote:
> Yep, that's the thing I need to do. Basically, it's on one of my servers
> and I need to run a Makefile. In the near term I can do that, but
> medium term we might need to find a way of hosting it that doesn't
> require it, 

Yeah!

> or share your SSH key to me so you can do it too? Should
> probably take this conversation out of the issue…

Yeah, I can share my SSH key! 

Perhaps you can email me at kjetil@kjernsmo.net, so we can coordinate?

Kjetil

**Comment by ewilderj** [comment]
@tobyink I can't find the schema for doap-deps on that page that @kjetilk linked
I did however find https://github.com/tobyink/rdf-doap-deps/blob/master/doap-deps.ttl

Do you have pans to re-host the Schema anywhere? Should we think about merging terms into DOAP?
